sbol2/partshop.py

```diff
@@ -323,3 +326,75 @@ def attachFile(self, top_level_uri, filepath):
         msg = 'HTTP Error code {} trying to attach file.'
         msg = msg.format(response.status_code)
         raise SBOLError(msg, SBOLErrorCode.SBOL_ERROR_BAD_HTTP_REQUEST)
+
+    def _make_search_item(self, item:dict) -> Identified:
```

Comment:
I think you might be able to leverage this for the advanced search -> the SearchQuery is essentially a dict-like object that allows the user to specify arbitrary search parameters (e.g., RDF predicates and objects)

Reply:
This method gets run on the response to build the Identified instances for return to the client. I think what you're talking about needs to go at the beginning when converting the SearchQuery to a URL to query SynBioHub